sdk/test-utils/recorder-new/src/utils/utils.ts

```diff
@@ -271,9 +270,46 @@ export function ensureExistence<T>(thing: T | undefined, label: string, mode: st
  *
  * If TEST_MODE is not defined, defaults to playback.
  */
-export function getTestMode(): string {
+export function getTestMode(): "record" | "playback" | "live" {
```

Comment:
Thinking maybe we should make this a named type and export it. Can imagine it getting some use in the tests.

Reply:
But TEST_MODE="random" is also valid and would be considered as playback.
So, exporting may not be mean much, if the users should be able to pass just strings or undefined.

Reply:
Shouldn't `getTestMode` return "playback" then in that case instead of "random"?

Reply:
I thought you meant exporting it to the public, can do for the tests in recorder-new

Reply:
Ah, so we're not actually exporting this `getTestMode` to the public at the moment? Then it's not such a big deal, I suppose.

Kind of tangential, but this function does feel kind of redundant with all the `is<x>Mode` helpers.

Reply:
is<x>Mode is meant for checking, getTestMode is meant for the case when we want the actual string.
For "undefined", isPlaybackMode() will return true, but we may want the string playback in some cases, for instance, letting the proxy-tool know through headers.

Reply:
hmm, not sure whether the isMode functions really add much then (since you _could_ just go `if(getTestMode() === "playback")` without them). But let's keep them in for consistency with the old recorder?

Reply:
Hmmm, makes sense. But yes, let's leave since we'd have to change a lot of places, not just the recorder.
Not worth the efforts even if we are doing major version upgrade

Reply:
`type TestMode = "record" | "playback" | "live";` seems simple, and you don't have to change it everywhere, but it's nice to have named instead of anonymous unions when you return only certain strings.